### PR TITLE
7.0 travel alarms

### PR DESCRIPTION
**Pull request by joaoalf**
This PR add email alerts for some events.

### CODE REVIEW

File: travel/res_config.py [hunk 1]
```diff
@@ -36,6 +38,26 @@ def get_basic_passenger_limit(ir_config_parameter_pool, cr, uid, context=None):
         return limit
 
 
+def get_alert_address(ir_config_parameter_pool, cr, uid, context=None):
+    context = context or {}
```
Comment (pedrobaeza) [inline travel/res_config.py:42]: Please use

```
if context is None:
    context = {}
```

or the more elaborated formula that @bwrsandman suggested a time ago.

Comment (bwrsandman) [inline travel/res_config.py:42]: `context = context or {}` does the same thing...
Why would this block the PR?

File: travel/travel.py [hunk 2]
```diff
@@ -164,7 +191,8 @@ def travel_reserve(self, cr, uid, ids, context=None):
         if type(ids) is not list:
             ids = [ids]
         for travel in self.browse(cr, uid, ids, context=context):
-            self.write(cr, uid, [travel.id], {'state': 'reserved'})
+            self.write(
+                cr, uid, [travel.id], {'state': 'reserved'})
```
Comment (bwrsandman) [inline travel/travel.py]: context not being passed here

File: travel/travel.py [hunk 3]
```diff
@@ -172,7 +200,8 @@ def travel_confirm(self, cr, uid, ids, context=None):
         if type(ids) is not list:
             ids = [ids]
         for travel in self.browse(cr, uid, ids, context=context):
-            self.write(cr, uid, [travel.id], {'state': 'confirmed'})
+            self.write(
+                cr, uid, [travel.id], {'state': 'confirmed'})
```
Comment (bwrsandman) [inline travel/travel.py]: context not being passed here

File: travel/res_config.py (hunk 1 above)
Comment (pedrobaeza) [inline travel/res_config.py:42]: Not exactly the same if the passed context is {}, you're overwriting the context.

Comment (bwrsandman) [inline travel/res_config.py:42]: With `{}`. The reference would be different, which means if something were later added to context, those changes wouldn't go back up the stack.

``` python
In [1]: context = {}

In [2]: id(context)
Out[2]: 140239108613576

In [3]: context = context or {}

In [4]: id(context)
Out[4]: 140239152358016

```

Is this what you mean?

Comment (joaoalf) [inline travel/res_config.py:42]: Hello @pedrobaeza 

Thank you for your coments. Can you elaborate on this context issue? I like very much the `context = context or {}` form and I've been using it for a couple of year now. Can you provide an example for your overwriting problem?

Comment (bwrsandman) [inline travel/res_config.py:42]:
``` python

class model(orm.Model):
    _name = "some.model"

    def foo(self, cr, uid, ids, context=None):
        if context is None:
            context = {}
        self.bar(cr, uid, ids, context=context)
        if context.get('some_key'):
             # do some stuff

    def bar(self, cr, uid, ids, context=None):
        if context is None:
            context = {}

class model_sub(orm.Model):
    _inherit = "some.model"

    def foo(self, cr, uid, ids, context=None):
        if context is None:
            context = {}
        super(model_sub, self).foo(cr, uid, ids, context=context)

    def bar(self, cr, uid, ids, context=None):
        context = context or {}
        context['some_key'] = True
        super(model_sub, self).bar(cr, uid, ids, context=context)
```

The trace goes as follows with `context = {}`:
- A call to `model_sub.foo` with `id(context) == 1`
  - A call to `model.foo` with `id(context) == 1`
    - A call to `model_sub.bar` with `id(context) == 1`
      - `context = context or {}` creates a new dict and assigns it to context `id(context) == 2`
      - A call to `model_sub.bar` with `id(context) == 2`
      - `context == {'some_key: True}`
- Return to `model.foo`
  - `id(context) == 1`
  - `context == {}`

**Comment by ehdem** [comment]
:+1:

**Comment by gurneyalex** [comment]
@joaoalf build is red on the last commit. Could you check what the matter is?

**Comment by bwrsandman** [comment]
:+1:

**Comment by pedrobaeza** [comment]
Please squash commits and check Travis status for this to be merged. If not interested anymore, you can close it.

**Comment by pedrobaeza** [comment]
As there's no answer, I close. Feel free to reopen in case there's interest.

**Pull request closed without merging**